### PR TITLE
[Chore] extracted table list common components

### PR DESCRIPTION
**Pull request by neighborhood999**
**PR checklist**

- [x] Ensure you have added or ran the appropriate tests for your PR.
- [x] DCO signed

### CODE REVIEW

File: static_report/src/components/shared/Tables/TableList/CRTableListItem/CRTableListAssertions.tsx [hunk 1]
```diff
@@ -249,45 +238,61 @@ export function CRTargetTableAssertionsSummary({
...
+          isPassed={isPassed}
+          isFailedEqual={isFailedEqual}
+          isMoreFailed={isMoreFailed}
+        />
+      }
+    />
+  );
+}
+
+function ComparisonLabelIcon({
```
Comment (jonycfu) [inline static_report/src/components/shared/Tables/TableList/CRTableListItem/CRTableListAssertions.tsx:258]: Would this be reusable from `components/Assertions/..` Just did a shallow search and found these

image
image
image

File: static_report/src/components/shared/Tables/TableList/CRTableListItem/index.tsx [hunk 2]
```diff
@@ -35,90 +31,65 @@ export function CRTableListItem({
...
-          <GridItem>
-            <Flex gap={10} color="gray.500">
-              <Text>Rows</Text>
-              <CRTableListDeltaSummary
-                baseCount={baseTableDatum?.row_count}
-                targetCount={targetTableDatum?.row_count}
-              />
+        <GridItem>
+          <TableItemName
+            name={columnName as string}
```
Comment (jonycfu) [inline static_report/src/components/shared/Tables/TableList/CRTableListItem/index.tsx:42]: suggest: handle `name` as `name?` in `TableItemName`

File: static_report/src/components/shared/Widgets/ColumnAssertionLabelWidget.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+export type ColumnAssertionLabelWidgetProps = {
+  total: number;
+  failed: number;
+  isPassed: boolean;
+  isComparison?: boolean;
+  icon?: ReactNode;
+  comparisonDelta?: ReactNode;
+};
+
+export function ColumnAssertionLabelWidget({
```
Comment (jonycfu) [inline static_report/src/components/shared/Widgets/ColumnAssertionLabelWidget.tsx:16]: I'm not sure this should be a widget, as it isn't a middle-level component that reuses our custom lower-level dumber components. 

I suggest this as a candidate for `components/shared/Assertions/..`

File: static_report/src/components/shared/Tables/TableList/SRTableListItem/SRTableListAssertionsSummary.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { ColumnAssertionLabelWidget } from '../../../Widgets/ColumnAssertionLabelWidget';
+import { getAssertions } from '../../../../../utils/assertion';
+import { type AssertionTest } from '../../../../../sdlc/single-report-schema';
+
+export function SRTableListAssertionsSummary({
```
Comment (jonycfu) [inline static_report/src/components/shared/Tables/TableList/SRTableListItem/SRTableListAssertionsSummary.tsx:5]: Is there a comparison version of this component? (couldn't find it xD)

If it's a standalone, I'd suggest removing the `SR` or `CR`, where possible.

Also, If it's general enough, I think this belongs in `components/shared/Assertions/..`

Comment (neighborhood999) [reply static_report/src/components/shared/Tables/TableList/SRTableListItem/SRTableListAssertionsSummary.tsx:5]: Good catch! I didn't notice, this component is redundant and can be replaced by another one.

File: static_report/src/components/shared/Tables/TableList/CRTableListItem/index.tsx (hunk 2 above)
Comment (neighborhood999) [reply static_report/src/components/shared/Tables/TableList/CRTableListItem/index.tsx]: Here i prefer to give it a default value.

**Review by jonycfu — APPROVED** [review]

**Pull request merged**